web/app/connectors/_sfcc-connector/account/commands.js

```
@@ -175,3 +179,57 @@ export const updateBillingAddress = (formValues) => (dispatch) => {
 export const initAccountDashboardPage = (url) => (dispatch) => { // eslint-disable-line
     return Promise.resolve()
 }
+
+/* eslint-disable camelcase */
+export const initAccountInfoPage = () => (dispatch) => {
+    const {sub} = getAuthTokenPayload()
+    const customerId = JSON.parse(sub).customer_info.customer_id
+    return makeApiJsonRequest(`/customers/${customerId}`)
+        .then(({first_name, last_name, login}) => {
+            const result = {
+                accountFormInfo: {
+                    names: `${first_name} ${last_name}`,
+                    email: login
+                }
+            }
+            dispatch(recieveAccountInfoUIData((result)))
+        })
+}
+
+
+export const updateAccountInfo = ({names, email}) => (dispatch) => {
+    const {sub} = getAuthTokenPayload()
+    const customerId = JSON.parse(sub).customer_info.customer_id
+    const requestBody = {
+        first_name: names.split(' ')[0],
+        last_name: names.split(' ')[1],
+        email
+    }
+
+    return makeApiJsonRequest(`/customers/${customerId}`, requestBody, {method: 'PATCH'})
+        .then(({first_name, last_name, login}) => {
+            const result = {
+                accountFormInfo: {
+                    names: `${first_name} ${last_name}`,
+                    email: login
+                }
+            }
+            dispatch(recieveAccountInfoUIData((result)))
+        })
+}
+
+export const updateAccountPassword = ({currentPassword, newPassword}) => (dispatch) => {
+    const {sub} = getAuthTokenPayload()
+    const customerId = JSON.parse(sub).customer_info.customer_id
+    const requestBody = {
+        current_password: currentPassword,
+        password: newPassword
+    }
+
+    return makeApiJsonRequest(`/customers/${customerId}/password`, requestBody, {method: 'PUT'})
+        .then(checkForResponseFault)
+        .catch(() => {
+            throw new SubmissionError({_error: 'Password Change Failed'})
```

I think we might want to look at what the fault is before just throwing an error. If it's an InvalidCustomerException then we should probably throw a different error, and handle that error differently in the action, possibly navigating the user to the login page since their login session has probably expired.